event_registration_cancel_reason/models/event_registration.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antiun Ingeniería S.L.
+# © 2016 Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import api, fields, models
+
+
+class EventRegistration(models.Model):
+    _inherit = 'event.registration'
+
+    cancel_reason_id = fields.Many2one(
+        comodel_name='event.registration.cancel.reason', readonly=True,
+        string="Reason for cancellation", ondelete="restrict")
+
+    @api.multi
+    def button_reg_cancel(self):
+        if self.env.context.get('bypass_reason'):
+            return super(EventRegistration, self).button_reg_cancel()
+        return {
+            'type': 'ir.actions.act_window',
+            'name': 'Cancel reason',
+            'res_model': 'event.registration.cancel.log.reason',
+            'view_type': 'form',
+            'view_mode': 'form',
+            'target': 'new',
+        }
+
+    @api.one
+    def do_draft(self):
```

Comment:
I'm afraid this is not something we can choose, because it's the name that Odoo gave it.